src/grimp/application/usecases.py

```diff
@@ -122,18 +150,25 @@ def _scan_packages(
         include_external_packages=include_external_packages,
     )
 
+    partial_get_imports_by_module = partial(
+        get_imports_by_module,
+        cache_dir=cache_dir,
+        import_scanner=import_scanner,
+        exclude_type_checking_imports=exclude_type_checking_imports,
+        cache=cache,
+    )
+
     for found_package in found_packages:
-        for module_file in found_package.module_files:
-            module = module_file.module
-            try:
-                if cache_dir is None:
-                    raise caching.CacheMiss
-                direct_imports = cache.read_imports(module_file)
-            except caching.CacheMiss:
-                direct_imports = import_scanner.scan_for_imports(
-                    module, exclude_type_checking_imports=exclude_type_checking_imports
-                )
-            imports_by_module[module] = direct_imports
+        with ProcessPoolExecutor() as executor:
+            chunk_size = ceil(len(found_package.module_files) / executor._max_workers) or 1
```

Comment:
Would be helpful to have a comment here explaining our reasoning for the chunk size.